src/components/Extensions/ArrayField.tsx

```diff
@@ -0,0 +1,265 @@
+import { Plus, Trash2 } from "lucide-react";
+import { Control, FieldValues, Path, useFieldArray } from "react-hook-form";
+import { useTranslation } from "react-i18next";
+
+import { cn } from "@/lib/utils";
+
+import { Button } from "@/components/ui/button";
+import { FormLabel } from "@/components/ui/form";
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from "@/components/ui/table";
+
+import { ExtensionFieldMetadata } from "@/Utils/schema/types";
+
+import { SchemaField } from "./SchemaField";
+
+interface ArrayFieldProps<TFieldValues extends FieldValues> {
+  /** Field metadata from JSON Schema */
+  metadata: ExtensionFieldMetadata;
+  /** React Hook Form control */
+  control: Control<TFieldValues>;
+  /** Base path for nested fields (e.g., "extensions") */
+  basePath?: string;
+  /** Additional class name for the container */
+  className?: string;
+}
+
+/**
+ * Renders an array field with add/remove functionality
+ * Supports table layout (x-ui: table) and list layout (x-ui: list)
+ */
+export function ArrayField<TFieldValues extends FieldValues>({
+  metadata,
+  control,
+  basePath,
+  className,
+}: ArrayFieldProps<TFieldValues>) {
+  const { t } = useTranslation();
+
+  const fieldPath = basePath
+    ? (`${basePath}.${metadata.name}` as Path<TFieldValues>)
+    : (metadata.name as Path<TFieldValues>);
+
+  const { fields, append, remove } = useFieldArray({
+    control,
+    name: fieldPath as never,
+  });
+
+  const itemMeta = metadata.itemMetadata;
+  const isTableLayout = metadata.uiControl === "table";
+  const nestedFields = itemMeta?.nestedFields || [];
+
+  // Get default values for new items
+  const getDefaultItem = () => {
+    if (itemMeta?.type === "object" && nestedFields.length > 0) {
+      const defaults: Record<string, unknown> = {};
+      for (const field of nestedFields) {
+        if (field.defaultValue !== undefined) {
+          defaults[field.name] = field.defaultValue;
+        }
+      }
+      return defaults;
+    }
+    return itemMeta?.defaultValue ?? {};
+  };
+
+  // Table layout for object arrays
+  if (isTableLayout && itemMeta?.type === "object" && nestedFields.length > 0) {
+    return (
+      <div className={cn("space-y-3", className)}>
+        <div className="flex justify-between items-center">
+          <FormLabel>
+            {metadata.label}
+            {metadata.required && <span className="text-red-500 ml-1">*</span>}
```

Comment:
thought we are switching to marking optional fields as such, instead of adding required or are we sticking with this